src/main/java/com/zaxxer/nuprocess/NuProcessBuilder.java

```diff
@@ -310,4 +310,26 @@ private String[] prepareEnvironment()
 
       return env;
    }
+
+   static int getGlobalBufferSizeSetting()
+   {
+      final String bufferCapacityProperty = System.getProperty(NuProcess.BUFFER_CAPACITY_PROPERTY);
+      if (bufferCapacityProperty == null || bufferCapacityProperty.trim().isEmpty())
+      {
+         return 64 * 1024;
+      }
+      try
+      {
+         final int value = Integer.parseInt(bufferCapacityProperty);
+         if (value < 1024 || value > 1024 * 1024)
+         {
+            throw new IllegalArgumentException("This -D" + NuProcess.BUFFER_CAPACITY_PROPERTY + "=" + value + " needs to be between 1 KiB and 1 MiB.");
+         }
+         return value;
+      }
+      catch (NumberFormatException e)
+      {
+         throw new IllegalArgumentException("This -D" + NuProcess.BUFFER_CAPACITY_PROPERTY + "=" + bufferCapacityProperty + " needs to be a number.");
+      }
+   }
```

Comment:
Please follow the code conventions of this project.
* Opening brace on same line
* For subsequent keywords (like `catch` or `else`) are on a separate line from the closing brace (as you have it)
* No unnecessary `final` qualifiers

Something like this:
```java
   static int getGlobalBufferSizeSetting() {
      String capacity = System.getProperty(NuProcess.BUFFER_CAPACITY_PROPERTY);
      if (capacity == null || capacity.trim().isEmpty()) {
         return 64 * 1024;
      }
      try {
         int value = Integer.parseInt(capacity);
         if (value < 1024) {
             // Log with JUL
             return 1024;
         }
         if (value > 1024 * 1024) {
             // Log with JUL
             return 1024 * 1024;
         }
         return value;
      }
      catch (NumberFormatException e) {
         // Log with JUL
         return 64 * 1024;
      }
   }
```